addon/components/ember-ckeditor.js

```diff
@@ -1,23 +1,130 @@
+  /**
+   * Primary value. Bindable.
+   * Still have a bit of magic to ensure events are fired when they should be
+   */
+  value: null,
+  /**
+   * Ferries any bound-in values into the editor.
+   */
+  _handleValueChange: Ember.observer('value', function () {
+    if (this._editorChanging) {
+      return;
+    }
+    if (this._editor) {
+      this._editor.setData(this.get('value'));
+    }
+  }),
 
-  didInsertElement() {
-    let textarea = this.element.querySelector('.editor');
-    let editor = this._editor = CKEDITOR.replace(textarea);
-    editor.on('change', (e) => {
-      this.sendAction('on-change', e.editor.getData());
-    });
+  /**
+   * Skin that should be loaded.
+   * Not Bindable.
+   */
+  skin: 'moono',
+  /**
+   * The content tags that are permitted in the editor. Null/default will let the plugins configure.
+   * Note: If you remove the toolbar, you cannot use automatic content, and MUST specify it.
+   * Not Bindable.
+   */
+  allowedContent: null,
+  /**
+   * The content tags that are not permitted, even if the auto-config says they're available.
+   * Not Bindable.
+   */
+  disallowedContent: null,
+  /**
+   * ReadOnly mode
+   */
+  readOnly: false,
+  /**
+   * Build the config passed to ckeditor's init method
+   * Ideal for over-writing in sub-classes
+   * @returns {} ckeditor config
+   */
+  buildConfig: function () {
+    var config = {
+      skin: this.get('skin'),
+      disallowedContent: this.get('disallowedContent'),
+      readOnly: this.get('readOnly'),
+      allowedContent: this.get('allowedContent')
+    };
+
+    return config;
   },
+  _handleDidInsertElement: Ember.on('didInsertElement', function () {
+    var editor = this._editor = this.$().ckeditor(this.buildConfig(), () => {
+      if (editor !== this._editor) {
+        //It's possible that the view re-rendered prior to the editor finished initializing.
+        //If so, destroy this editor.
+        editor.destroy(true);
+      } else {
+        editor.setData(this.get('value'), () => {
+          this.trigger('editor-ready');
+        });
+      }
+    }).editor;
+    this._editor.on('change', this._handleChange.bind(this));
+  }),
+  _handleWillClearRender: Ember.on('willClearRender', function () {
+    if (this._editor) {
+      if (this._editor.status === 'ready') {
+        //We can only destroy an editor that is ready.
+        //We check for orphaned editors in the insertElement callback
+        this._editor.destroy(true);
+      }
+      this._editor = null;
+    }
+  }),
+  /**
+   * Handles ckeditor change events, props them into the value object, sends actions and events.
+   * @param event
+   * @private
+   */
+  _handleChange: function (event) {
+    //We set a flag so we can disregard changes to 'value'
+    this._editorChanging = true;
+    try {
+      var newValue = event.editor.getData();
 
-  willDestroyElement() {
-    this._editor.destroy();
-    this._editor = null;
+      this.set('value', newValue);
```

Comment:
I don't care too much about the two-way binding, but any code not following strict data-down/actions-up will need to be able to retrieve the value in some way. We can't prevent lib users from reading 'value', which means that they might be very surprised that the value never changes.

I've worked through several implementations of this, and this was the simplest way to support our current non-DDAU workflow.